packages/yuudachi/src/util/generateHistory.ts

```diff
@@ -136,34 +111,203 @@ export async function generateHistory(
 		...embed,
 	};
 
-	const summary: string[] = [];
-	let truncated = false;
+	// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+	if (cases.length > 0) {
+		const summary: string[] = [];
+		let truncated = false;
+
+		for (const c of cases) {
+			const dateFormatted = time(dayjs(c.created_at).unix(), TimestampStyles.ShortDate);
+			const caseString = `${dateFormatted} ${inlineCode(`${ACTION_KEYS[c.action]!.toUpperCase()}`)} ${
+				c.log_message_id
+					? hyperlink(`#${c.case_id}`, messageLink(modLogChannelId, c.log_message_id), c.guild_id)
+					: `#${c.case_id}`
+			} ${c.reason?.replace(/\*/g, '') ?? ''}`;
 
-	for (const c of cases) {
-		const dateFormatted = time(dayjs(c.created_at).unix(), TimestampStyles.ShortDate);
-		const caseString = `${dateFormatted} ${inlineCode(`${ACTION_KEYS[c.action]!.toUpperCase()}`)} ${
-			c.log_message_id
-				? hyperlink(`#${c.case_id}`, messageLink(modLogChannelId, c.log_message_id), c.guild_id)
-				: `#${c.case_id}`
-		} ${c.reason?.replace(/\*/g, '') ?? ''}`;
+			if (summary.join('\n').length + caseString.length + 1 < 4060) {
+				summary.push(caseString);
+				continue;
+			}
 
-		if (summary.join('\n').length + caseString.length + 1 < 4060) {
-			summary.push(caseString);
-			continue;
+			truncated = true;
+			break;
 		}
 
-		truncated = true;
-		break;
+		if (truncated) {
+			embed = {
+				description: i18next.t('log.history.summary_truncated', { summary: summary.join('\n'), lng: locale }),
+				...embed,
+			};
+		} else {
+			embed = { description: summary.join('\n'), ...embed };
+		}
+	} else {
+		embed = {
+			description: i18next.t('log.history.none', { lng: locale }),
+			...embed,
+		};
 	}
 
-	if (truncated) {
+	return truncateEmbed(embed);
+}
+
+export async function generateReportHistory(
+	interaction: CommandInteraction<'cached'> | ButtonInteraction<'cached'> | SelectMenuInteraction<'cached'>,
+	target: { user: User; member?: GuildMember | undefined },
+	locale: string,
+	omitAuthor = false,
+) {
+	const sql = container.resolve<Sql<any>>(kSQL);
+	const reportChannelId = await getGuildSetting(interaction.guildId, SettingsKeys.ReportChannelId);
+
+	let embed = addFields({
+		author: omitAuthor
+			? undefined
+			: {
+					name: `${target.user.tag} (${target.user.id})`,
+					icon_url: target.user.displayAvatarURL(),
+			  },
+		title: i18next.t('log.history.report_title', { lng: locale }),
+	});
+
+	const targetReports = await sql<[RawReport]>`
+			select *
+			from reports
+			where guild_id = ${interaction.guildId}
+				and target_id = ${target.user.id}
+				and status not in (${ReportStatus.Rejected}, ${ReportStatus.False})
+			order by created_at desc`;
+
+	const falseReports = await sql<[RawReport]>`
```

Comment:
Should probably avoid multiple IO calls and just do the partitioning in code.

Reply:
Hmmm, will test some alternatives, tyy

Reply:
Fixed in 5e9cf1d